common/src/main/java/com/wynntils/mc/mixin/CapeLayerMixin.java

```diff
@@ -40,4 +53,32 @@ private void render(
             ci.cancel();
         }
     }
+
+    @ModifyArg(
+            method = "render",
+            at =
+                    @At(
+                            value = "INVOKE",
+                            target =
+                                    "net/minecraft/client/renderer/MultiBufferSource.getBuffer(Lnet/minecraft/client/renderer/RenderType;)Lcom/mojang/blaze3d/vertex/VertexConsumer;"))
+    private RenderType setTranslucenceCapeRenderType(
+            RenderType original,
+            @Local(argsOnly = true) AbstractClientPlayer livingEntity,
+            @Local PlayerSkin playerSkin) {
+        Minecraft minecraft = Minecraft.getInstance();
+        boolean isBodyVisible = !livingEntity.isInvisible();
+        boolean translucent = !isBodyVisible && !livingEntity.isInvisibleTo(minecraft.player);
+
+        LivingEntityRenderTranslucentCheckEvent event =
+                new LivingEntityRenderTranslucentCheckEvent(translucent, livingEntity, translucent ? 0.15f : 1.0f);
+        MixinHelper.post(event);
+
+        // Translucence value needs to pass into PlayerModel class, because renderCloak method has no 'color' argument
+        PlayerModelExtension playerModelExtension = (PlayerModelExtension) this.getParentModel();
```

Comment:
Referring to my earlier comment;

Would you think it's feasible change to make the event itself hold the PlayerModelExtension, and the event users can modify the translucence (and perhaps, in the future, other specific settings) directly, without the event acting as kind of a middle man.